osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/application/mapping/DeviceConverter.java

```diff
@@ -33,6 +33,14 @@ public DlmsDevice convertTo(
     dlmsDevice.setMbusManufacturerIdentification(source.getMbusManufacturerIdentification());
     dlmsDevice.setProtocol(source.getProtocolName(), source.getProtocolVersion());
     dlmsDevice.setTimezone(source.getTimezone());
+    dlmsDevice.setIpAddressIsStatic(source.isIpAddressIsStatic());
+    dlmsDevice.setWithListSupported(source.isWithListSupported());
+    dlmsDevice.setSelectiveAccessSupported(source.isSelectiveAccessSupported());
+    dlmsDevice.setPolyphase(source.isPolyphase());
+    dlmsDevice.setPort(source.getPort());
+    if (source.getChallengeLength() != null && source.getChallengeLength() > 0) {
+      dlmsDevice.setChallengeLength(source.getChallengeLength());
```

Comment:
The [installation-ws-smartmetering.xsd](https://github.com/OSGP/open-smart-grid-platform/pull/980/files#diff-92569c86c9ff67f62fb928f8a2dc6cad0fe99f1f3342365112ef01693863f8ca) defines both attributes not optional
```
      <xsd:element name="ChallengeLength" type="xsd:int" />
      <xsd:element name="Port" type="xsd:long" />
```
Why not do a null check for the port like you do for ChallengeLength?